nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment-api/src/main/java/org/nuxeo/ecm/platform/comment/api/CommentManager.java

```diff
@@ -322,4 +324,35 @@ void deleteExternalComment(CoreSession session, String entityId)
      */
     boolean hasFeature(Feature feature);
 
+    /**
+     * Gets the top level ancestor document ref for the given document model comment. No matter how many levels of
+     * comments there is.
+     * <p>
+     * Given a document fileOne, that we comment with commentOne which we reply on with replyOne
+     * <p>
+     * This method will return:
+     * <ul>
+     * <li>CommentManager#getAncestorRef(session, commentOne) = fileOne</li>
+     * <li>CommentManager#getAncestorRef(session, replyOne) = fileOne</li>
+     * </ul>
+     *
+     * @param session the CoreSession
+     * @param documentModel the comment document model
+     * @return the top level ancestor document ref
+     * @since 11.1
+     */
+    default DocumentRef getAncestorRef(CoreSession session, DocumentModel documentModel) {
```

Comment:
This one is not needed in the interface, it could exists in implementation with protected scope.

Btw, great javadoc :+1: